### PR TITLE
ipv6 ping test included as a part of network test

### PR DESCRIPTION
**Pull request by vaishnavibhat**
Fetch ipv6 address for host and peer and run the ping test to see if its success.

Signed-off-by: Vaishnavi Bhat <vaishnavi@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by vaishnavibhat** [comment]
avocado run --test-runner runner network_test.py:NetworkTest.test_ipv6_ping -m network_test.yaml 
JOB ID     : 9dc6a0e0a46072a2b387d32cd96d746f29c4bb47
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2022-10-21T12.05-9dc6a0e/job.log
 (1/9) network_test.py:NetworkTest.test_ipv6_ping;run-mtu-1500-1870: PASS (15.43 s)
 (2/9) network_test.py:NetworkTest.test_ipv6_ping;run-mtu-2000-1e02: PASS (16.62 s)
 (3/9) network_test.py:NetworkTest.test_ipv6_ping;run-mtu-3000-26ed: PASS (15.48 s)
 (4/9) network_test.py:NetworkTest.test_ipv6_ping;run-mtu-4000-5bf7: PASS (16.19 s)
 (5/9) network_test.py:NetworkTest.test_ipv6_ping;run-mtu-5000-e530: PASS (18.38 s)
 (6/9) network_test.py:NetworkTest.test_ipv6_ping;run-mtu-6000-b497: PASS (18.71 s)
 (7/9) network_test.py:NetworkTest.test_ipv6_ping;run-mtu-7000-31c4: PASS (18.30 s)
 (8/9) network_test.py:NetworkTest.test_ipv6_ping;run-mtu-8000-2e9c: PASS (17.98 s)
 (9/9) network_test.py:NetworkTest.test_ipv6_ping;run-mtu-9000-5709: PASS (17.61 s)
RESULTS    : PASS 9 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2022-10-21T12.05-9dc6a0e/results.html
JOB TIME   : 164.14 s

[ipv6.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/9840715/ipv6.log)

The entire network_test.py runs successfully with the new test added.

File: io/net/network_test.py [hunk 1]
```diff
@@ -191,6 +191,23 @@ def test_floodping(self):
...
+            self.cancel("IPV6 addrress is not set for host interface")
+        try:
+            peer_ipv6 = self.peer_networkinterface.get_ipaddrs(version=6)
+            if not peer_ipv6[0]:
+                self.cancel("IPV6 addrress is not set for peer interface")
+        except Exception:
+            self.cancel("Test failing while getting IPV6 address for peer interface")
+        if self.networkinterface.ping_check(peer_ipv6[0], count=10) is not None:
+            self.fail("IPV6 ping test failed")
+
```
Comment (abdhaleegit) [inline io/net/network_test.py:210]: @vaishnavibhat are we not going to pass the IPV6 IP addr as input and configure the same before getting and doing ping_check?

**Review by abdhaleegit — CHANGES_REQUESTED** [review]

**Review by vaishnavibhat — COMMENTED** [review]
@abdhaleegit Currently the switch and the cfg configuration is assigning ipv6 automatically to the interfaces. The tests tries to extract this value on both host and peer and do a ping test.
Manual assignment is not recommended but if we want to extend ip addr add/ del functionality with the automatically assigned ipv6, then I can extend this test to include those. Not going to ask the user to give any values in yaml file, will be reading the ipv6 thats already assigned

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Pull request merged**